otpush/push.py

```diff
@@ -225,12 +225,27 @@ def start(self):
             raise OTPushError(
                 f'SSH control socket {self.socket} does not exist')
 
+    def wait_for_exit(self, timeout):
+        try:
+            self.master_proc.wait(timeout)
+            return True
+        except subprocess.TimeoutExpired:
+            return False
+
     def stop(self):
+        wait_timeout = 20
         if self.master_proc is not None:
             if self.master_proc.poll() is None:
                 logger.debug('Stopping SSH master process %d',
                              self.master_proc.pid)
                 self.master_proc.terminate()
+                if self.wait_for_exit(wait_timeout):
+                    return
+                self.master_proc.kill()
+                if self.wait_for_exit(wait_timeout):
+                    return
+                logger.debug('Failed to stop the SSH master process %d',
```

Comment:
I think this should be an `error` level message. If the SSH server can't be killed with `SIGKILL` and is therefore a zombie, I think you'd want to see that message loudly.